core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlUtils.java

```diff
@@ -19,46 +19,30 @@
 import org.apache.calcite.rel.RelNode;
 import org.apache.calcite.rel.core.Filter;
 import org.apache.calcite.rel.core.Project;
+import org.apache.calcite.rel.logical.LogicalJoin;
 import org.apache.calcite.rex.RexCall;
 import org.apache.calcite.rex.RexInputRef;
-import org.apache.calcite.rex.RexLiteral;
 import org.apache.calcite.rex.RexNode;
 import org.apache.calcite.rex.RexOver;
-
-import java.util.ArrayList;
-import java.util.List;
+import org.apache.calcite.rex.RexVisitorImpl;
+import org.apache.calcite.util.Util;
 
 /**
  * Utility class for rel2sql package.
  */
 public class RelToSqlUtils {
 
-  /** Returns list of all RexInputRef objects from the given condition. */
-  private List<RexNode> getOperandsOfTypeRexInputRefFromRexNode(RexNode conditionRex,
-      List<RexNode> inputRefRexList) {
-    if (conditionRex instanceof RexInputRef) {
-      inputRefRexList.add(conditionRex);
-    } else if (conditionRex instanceof RexCall) {
-      for (RexNode operand : ((RexCall) conditionRex).getOperands()) {
-        if (operand instanceof RexLiteral) {
-          continue;
-        } else {
-          getOperandsOfTypeRexInputRefFromRexNode(operand, inputRefRexList);
-        }
-      }
-    }
-    return inputRefRexList;
-  }
-
-  /** Returns whether an operand is Analytical Function by traversing till next project rel
+  /**
+   * Returns whether an operand is Analytical Function by traversing till next project rel
    * For ex, FilterRel e1 -> FilterRel e2 -> ProjectRel p -> TableScan ts
    * Here, we are traversing till ProjectRel p to check whether an operand of FilterRel e1
-   * is Analytical function or not. */
-  private boolean isOperandAnalyticalInFollowingProject(RelNode rel, Integer rexOperandIndex) {
+   * is Analytical function or not.
+   */
+  private static boolean isOperandAnalyticalInFollowingProject(RelNode rel,
+      Integer rexOperandIndex) {
     if (rel instanceof Project) {
       return (((Project) rel).getProjects().size() - 1) >= rexOperandIndex
-          && (((Project) rel).getProjects().get(rexOperandIndex) instanceof RexOver
-          || isRexOverPresentInRexCall(((Project) rel).getProjects().get(rexOperandIndex)));
+          && isAnalyticalRex(((Project) rel).getProjects().get(rexOperandIndex));
     } else {
       if (rel.getInputs().size() > 0) {
```

Comment:
can we use else if here as below : 

else if (!rel.getInputs().isEmpty()) {
        return isOperandAnalyticalInFollowingProject(rel.getInput(0), rexOperandIndex);
    }

Reply:
Changed it.Please have a look